pkg/sendbuffer/sendbuffer.go

```diff
@@ -30,6 +30,9 @@ type SendBuffer struct {
 	sender                                      sender
 	sendTicker                                  *time.Ticker
 	isSending                                   bool
+
+	// logsJustPurged is used to prevent attempting to delete logs that were just purged
+	logsJustPurged bool
```

Comment:
is this safe as bool and not atomic.Bool because we think it's adequately protected by the writeMutex? (it seems like it at first glance just wanted to double check)

Reply:
yes, it's protected by the writeMutex... trying to think of a better way to do this, but I think this will suffice